datayoga-py/datayoga/producer.py

```diff
@@ -1,18 +1,19 @@
 import logging
-from typing import Any, Dict, List, TypedDict
+from typing import Any, Dict, Generator, List, TypedDict
 
 from .block import Block
 
 logger = logging.getLogger("dy")
+Message = TypedDict("Message", msg_id=str, value=Dict[str, Any])
 
 
 class Producer(Block):
 
-    def produce(self) -> List[TypedDict("Message", msg_id=str, value=Dict[str, Any])]:
+    async def produce(self) -> Generator[Message, None, None]:
```

Comment:
why does the produce need to be async?

Reply:
because this in the std.read producer:
```
data = await aioconsole.ainput()
```

Reply:
let's find out why producer needed to be async

Reply:
reverted